src/Infrastructure/Services/GenericSmsService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.Extensions.Options;
+using MRA.Jobs.Application.SMSService;
+
+namespace MRA.Jobs.Infrastructure.Services;
+
+public class GenericSmsService : ISmsService
+{
+    private readonly GenericSmsSettings _genericSmsSettings;
+
+    public GenericSmsService(IOptions<GenericSmsSettings> genericSmsSettings)
+    {
+        _genericSmsSettings = genericSmsSettings.Value;
+    }
+
+    public async Task<string> SendSmsAsync(SmsMessage message)
```

Comment:
This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.